packages/makeswift-next-plugin/index.js

```diff
@@ -58,6 +58,31 @@ module.exports =
           fallback: Array.isArray(rewrites) ? [] : rewrites?.fallback ?? [],
         }
       },
+      async headers() {
+        const headers = await nextConfig.headers?.()
+
+        return [
+          ...headers,
+          {
+            source: '/:path*',
+            has: [
+              {
+                type: 'query',
+                key: 'x-makeswift-load-fonts',
```

Comment:
This rewrite seems to apply to all files and doesn't see to do anything font specific, so why make the query parameter `x-makeswift-load-fonts`?

Also, what's to prevent anybody that's sending a request to the Next.js app from using this query parameter?

And, will there be a conflict with us adding this header if there's some functionality in Next.js or perhaps Vercel that sets the `Access-Control-Allow-Origin` header to a different value, say `*`?

Reply:
> This rewrite seems to apply to all files and doesn't see to do anything font specific, so why make the query parameter `x-makeswift-load-fonts`?

Yes. I totally agree with this. I was thinking to make a generic query that enables CORS for `app.makeswift.com`. Updated to `x-makeswift-allow-origin`.

> Also, what's to prevent anybody that's sending a request to the Next.js app from using this query parameter?

Nothing. Everyone can use this query param. This should be safe because we're setting the `Access-Control-Allow-Origin` to be `app.makeswift.com`, and not `*`. So, even if someone is using this query from other origin, it won't be allowed, because we're just allowing `app.makeswift.com` and not the request origin.

> And, will there be a conflict with us adding this header if there's some functionality in Next.js or perhaps Vercel that sets the Access-Control-Allow-Origin header to a different value, say *?

I thought about it. But, we're only setting the `Access-Control-Allow-Origin` header only on a request with a special query param, so it shouldn't override any existing functionality. 

Users can still intentionally or unintentionally override this behavior by setting the `Access-Control-Allow-Origin` header on their `next.config.js` file. I don't think there's a way to prevent that.

Reply:
Ok, this all sounds good! Let's go ahead with it and gather data/learn if there's any issues